reactReview/first-react-project/src/component/header/Search.js

```diff
@@ -0,0 +1,46 @@
+import React, {Component} from "react";
+import styles from './style/SearchStyle.module.css'
+import {withRouter} from "react-router-dom";
+
+class Search extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {searchInput: ''}
+        this.changeSearch = this.changeSearch.bind(this);
+        this.buttonClicked = this.buttonClicked.bind(this);
+    }
+
+    render() {
+        const searchClass = `material-icons ${styles.searchIcon}`
+        return (
+            <div className={styles.search}>
+                <input placeholder="Find File" onChange={this.changeSearch} className={styles.searchField}/>
+                <button onClick={this.buttonClicked} className={styles.searchButton}>
+                    search
+                    <span className={searchClass}>search</span>
+                </button>
+            </div>
+        )
+    }
+
+    changeSearch(event) {
+        this.setState({searchInput: event.target.value});
+    }
+
+    buttonClicked() {
+        if (sessionStorage.getItem('token') === null) {
+            alert('Login first');
+        } else {
+            let url= sessionStorage.getItem('url');
+            url = url.split('/');
+            url = url[url.length - 1];
+            url = url.substring(0, 1);
+            const mask = this.state.searchInput;
+            url = `files/search/${url}?mask=${mask}`;
+            sessionStorage.setItem('url', url);
+            this.props.history.go(0);
+        }
```

Comment:
else block looks like a function

Reply:
honestly it looks super unclear

Reply:
do you know what HATEOAS is?
I guess it would be more sense to construct urls on back-end side.
Read about that. But I dont think we have time for rework.